### DISCUSSION TITLE
How to use with http.ServeFile?

### DISCUSSION THREAD

**Discussion by gregoryjjb** [swaggest/rest#187] [Q&A] [answered]
I have an endpoint that returns large files of varying Content-Types. The regular `http.ServeFile` is very convenient because it handles range requests and sets the Content-Type automatically. Inside a `usecase.Interactor` function, can I somehow get access to the regular `http.ResponseWriter` and `http.Request` so I can use `http.ServeFile`?

**Comment by vearutop** [comment, marked as answer]
Hi, the easiest way to access those would be to embed (or use directly if you don't need extra parameters/headers) `request.EmbeddedSetter` and `response.EmbeddedSetter` in your in/out.

Please check an example.
https://github.com/vearutop/photo-blog/blob/v0.0.4/internal/usecase/serve_site_file.go
```go
package usecase

import (
	"context"
	"net/http"
	"path"

	"github.com/swaggest/rest/request"
	"github.com/swaggest/rest/response"
	"github.com/swaggest/usecase"
)

func ServeSiteFile(deps interface{}) usecase.Interactor {
	type fileReq struct {
		File string `path:"file"`
		request.EmbeddedSetter
	}

	u := usecase.NewInteractor(func(ctx context.Context, in fileReq, out *response.EmbeddedSetter) error {
		rw := out.ResponseWriter()

		rw.Header().Set("Cache-Control", "max-age=31536000")

		filePath := path.Join("site", in.File)

		http.ServeFile(rw, in.Request(), filePath)

		return nil
	})
	u.SetTags("Site")

	return u
}
```

**Reply by gregoryjjb** [reply]
Works perfectly, thank you!
